src/hooks/useHasAppUpdated/useHasAppUpdated.ts

```diff
@@ -3,45 +3,79 @@ import axios from 'axios'
 import isEqual from 'lodash/isEqual'
 import { useMemo, useState } from 'react'
 
-const APP_UPDATE_CHECK_INTERVAL = 1000 * 60
+export const APP_UPDATE_CHECK_INTERVAL = 1000 * 60
 
+const fetchAsset = async (url: string) => {
+  try {
+    // dummy query param bypasses browser cache
+    const { data } = await axios.get(`${url}?${new Date().valueOf()}`)
+    return data
+  } catch (e) {
+    console.error(`useHasAppUpdated: error fetching ${url}`, e)
+  }
+}
 export const useHasAppUpdated = () => {
   const [hasUpdated, setHasUpdated] = useState(false)
   const [initialManifestMainJs, setManifestMainJs] = useState(null)
+  const [initialEnv, setEnv] = useState(null)
   // asset-manifest tells us the latest minified built files
-  const url = '/asset-manifest.json'
-  const storeMainManifestJs = async () => {
-    // dummy query param bypasses browser cache
-    const { data } = await axios.get(`${url}?${new Date().valueOf()}`)
-    setManifestMainJs(data)
-  }
-  useMemo(storeMainManifestJs, [])
-  useInterval(async () => {
+  const assetManifestUrl = '/asset-manifest.json'
+  const envUrl = '/env.json'
+
+  const storeAndCompareAsset = async () => {
     // we don't care about updates locally obv
     if (window.location.hostname === 'localhost') return
 
-    let manifestMainJs
-    try {
-      const { data } = await axios.get(`${url}?${new Date().valueOf()}`)
-      manifestMainJs = data
-    } catch (e) {
-      console.error(`useHasAppUpdated: error fetching asset-manifest.json`, e)
-    }
+    const manifestMainJs = await fetchAsset(assetManifestUrl)
+    const env = await fetchAsset(envUrl)
+
     if (!manifestMainJs) {
       console.error(`useHasAppUpdated: can't find main.js in asset-manifest.json`)
-      return
     }
-    //deep equality check
-    let isSameAssetManifest = isEqual(initialManifestMainJs, manifestMainJs)
-    if (!isSameAssetManifest) {
+    if (!env) {
+      console.error(`useHasAppUpdated: can't find env in env.json`)
+    }
+
+    let isSameAssetManifest = true,
+      isSameEnv = true
+
+    if (!initialManifestMainJs) {
+      // first run
+      if (manifestMainJs) setManifestMainJs(manifestMainJs)
+    } else {
+      // subsequent runs
+      if (manifestMainJs && !isEqual(manifestMainJs, initialManifestMainJs)) {
+        isSameAssetManifest = false
+      }
+    }
+
+    if (!initialEnv) {
+      // first run
+      if (env) setEnv(env)
+    } else {
+      // subsequent runs
+      if (env && !isEqual(env, initialEnv)) {
+        isSameEnv = false
+      }
+    }
```

Comment:
hey @trungtin  - as per @0xApotheosis comments, please don't use this let and mutate pattern. 

here's an example of how to do it better

https://github.com/shapeshift/lib/blob/main/packages/investor-foxy/src/api/api.ts#L549

Reply:
> hey @trungtin - as per @0xApotheosis comments, please don't use this let and mutate pattern.
> 
> here's an example of how to do it better
> 
> https://github.com/shapeshift/lib/blob/main/packages/investor-foxy/src/api/api.ts#L549

You might want to link to a specific commit and not main. The link right now goes to this code:

```ts
const payload = {
      bip44Params,
      chainId,
      data,
      estimatedGas,
      gasPrice,
      nonce,
      to: contractAddress,
      value: '0'
    }
```
which doesn't seem relevant.